### ISSUE TITLE
[Edge] Folder creation using the web ui fails.

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/core#21710]
### Steps to reproduce

Using Microsoft's Edge do:

1. Create a folder using the web UI.

### Expected behaviour
Folder is created.

### Actual behaviour
There is an error. MKCOL operation fails.

### Server configuration
**Operating system**:
CentOS 6.6

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
Master branch
{"installed":true,"maintenance":false,"version":"9.0.0.6","versionstring":"9.0 pre alpha","edition":""}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
```
Enabled:
  - activity: 2.2.1
  - admin_audit: 0.7
  - dav: 0.1.3
  - enterprise_key: 0.1.0
  - files: 1.4.1
  - files_pdfviewer: 0.8
  - files_sharing: 0.8.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoviewer: 0.1.3
  - firewall: 2.3.0
  - firstrunwizard: 1.1
  - gallery: 14.3.0
  - notifications: 0.2.2
  - provisioning_api: 0.4.0
  - templateeditor: 0.1
  - windows_network_drive: 0.1.1
Disabled:
  - encryption
  - external
  - federation
  - files_antivirus
  - files_drop
  - files_external
  - files_ldap_home
  - objectstore
  - password_policy
  - sharepoint
  - user_external
  - user_ldap
  - user_shibboleth
```

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** 
No

**Logs**

```
{"reqId":"oR\/RcedmbCUBhI8MJjpe","remoteAddr":"176.31.107.195","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 415 The request body for the MKCOL request must have an xml Content-Type\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpMkcol(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:MKCOL', Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(53): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/remote.php(137): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#6 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php\",\"Line\":563}","level":4,"time":"2016-01-14T10:08:57+00:00"}
```

### Client configuration

**browser**
Edge

**Comment by MorrisJobke** [comment]
It simply doesn't create the folder and the response to the MKCOL request is "415 Unsupported Media Type":

![bildschirmfoto 2016-01-26 um 10 13 06](https://cloud.githubusercontent.com/assets/245432/12576654/9144e1dc-c415-11e5-82b0-7ddcf6e2b359.png)

A notification with `Could not create folder "asdasb"` pops up. The same works fine in Safari or Chrome.

**Comment by rullzer** [comment]
Ok found out why.

https://github.com/owncloud/core/blob/master/core/vendor/davclient.js/lib/client.js#L175

For mkcol the body is undefined. however if you send that in edge it still creates a real request with contentype and length... which is when sabre says... NO!!!

**Comment by rullzer** [comment]
Basically a bug in the davclient we use. PR for that in https://github.com/evert/davclient.js/issues/17

**Comment by norcalhelpdesk** [comment]
I have the same problem too.  I installed the software and couldn't create folders in Internet Explorer.  Works fine on Chrome.  Today is 4/6/16 and I installed the latest version on 14.04.1-Ubuntu .  I was using IE 11 version 11.0.9600.18231

also when I share a file, The password part is blank so people don't know what to put in there.

![image](https://cloud.githubusercontent.com/assets/18315999/14332828/3b3ccc96-fc00-11e5-96ef-8329ca818607.png)
